src/neural/loader.cc

```diff
@@ -48,14 +48,97 @@ namespace lczero {
 namespace {
 const std::uint32_t kWeightMagic = 0x1c0;
 
+uint32_t read_le(const uint8_t *addr) {
+  return addr[0] + 256 * addr[1] + 65536 * addr[2] + 16777216 * addr[3];
+}
+
+// Read a .zip file containing a network file (just stored, not compressed a
+// second time) that is appended at the end of the lc0 executable. Such a zip
+// file can be generated using, for example, "zip -0 zipfile.zip net.pb.gz",
+// "7z a -mx=0 zipfile.zip net.pb.gz" or the equivalent options of any other
+// compression utility.
+
+std::string DecompressEmbedded(std::string str) {
+  constexpr int MOD_GZIP_ZLIB_WINDOWSIZE = 15;
```

Comment:
Using zip looks like a huge overcomplication. Why not just append the file + file length + 4-byte magic?